Microsoft.Toolkit.Uwp.UI/Extensions/TextBox/TextBoxExtensions.SurfaceDial.cs

```diff
@@ -2,330 +2,131 @@
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <returns>Return value of property</returns>
-        public static double GetMinValue(TextBox obj)
-        {
-            return (double)obj.GetValue(MinValueProperty);
-        }
-
-        /// <summary>
-        /// Setter of the MinValue
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <param name="value">The value to set the property to.</param>
-        public static void SetMinValue(TextBox obj, double value)
-        {
-            obj.SetValue(MinValueProperty, value);
-        }
-
-        /// <summary>
-        /// Setter of the StepValue.
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <returns>Return value of property</returns>
-        public static double GetStepValue(TextBox obj)
-        {
-            return (double)obj.GetValue(StepValueProperty);
-        }
-
-        /// <summary>
-        /// Getter of the StepValue
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <param name="value">The value to set the property to.</param>
-        public static void SetStepValue(TextBox obj, double value)
-        {
-            obj.SetValue(StepValueProperty, value);
-        }
-
-        /// <summary>
-        /// Getter of the Icon
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <returns>Return value of property</returns>
-        public static RadialControllerMenuKnownIcon GetIcon(TextBox obj)
-        {
-            return (RadialControllerMenuKnownIcon)obj.GetValue(IconProperty);
-        }
-
-        /// <summary>
-        /// Setter of the Icon
-        /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <param name="value">The value to set the property to.</param>
-        public static void SetIcon(TextBox obj, RadialControllerMenuKnownIcon value)
-        {
-            obj.SetValue(IconProperty, value);
-        }
-
         /// <summary>
-        /// Setter of the Haptic Feedback property
+        /// The <see cref="SurfaceDialOptions"/> instance containing properties to configure the Surface Dial support for a <see cref="TextBox"/>.
         /// </summary>
-        /// <param name="obj">The Dependency Object we are dealing with, like a TextBox.</param>
-        /// <returns>Return value of property</returns>
-        public static bool GetEnableHapticFeedback(TextBox obj)
-        {
-            return (bool)obj.GetValue(EnableHapticFeedbackProperty);
-        }
+        public static readonly DependencyProperty SurfaceDialOptionsProperty = DependencyProperty.RegisterAttached(
```

Comment:
Should the property just be `SurfaceDial` which takes the `SurfaceDialOptions` or are you thinking it's better to have them named for consistency?

Reply:
So my rationale here was that "SurfaceDial" makes me thing the property _is_ a Surface Dial, as in, somehow assigning a Surface Dial to a `TextBox`, rather than options _for_ a Surface Dial. So using "SurfaceDialOptions" made it clearer when just reading through the property, as it immediately makes the intent clear and it's unambiguous. Thoughts? 🙂